payment-method-messaging/src/main/java/com/stripe/android/paymentmethodmessaging/view/PaymentMethodMessagingView.kt

```diff
@@ -0,0 +1,356 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.AbstractComposeView
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.intl.Locale
+import androidx.compose.ui.unit.dp
+import androidx.core.text.HtmlCompat
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.ViewModelStoreOwner
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.stripe.android.uicore.image.StripeImageLoader
+import com.stripe.android.uicore.text.EmbeddableImage
+import com.stripe.android.uicore.text.Html
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import com.stripe.android.paymentmethodmessaging.view.theme.Color as PaymentMethodMessageColor
+
+/**
+ * A view that displays promotional text and images for payment methods like Afterpay and Klarna.
+ * For example, "As low as 4 interest-free-payments of $9.75". When tapped, this view presents a
+ * full-screen Custom Chrome Tab to the customer with additional information ont he payment methods
+ * being displayed.
+ *
+ * You can embed this into your checkout or product screens to promote payment method options to
+ * your customer.
+ *
+ * Note: You must initialize this view with [PaymentMethodMessagingView.load]. For example:
+ *
+ * PaymentMethodMessagingView.load(
+ *     config = config,
+ *     onSuccess = {
+ *         // Show view
+ *     },
+ *     onFailure = {
+ *         // Show error
+ *     }
+ * )
+ */
+internal class PaymentMethodMessagingView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : AbstractComposeView(context, attrs, defStyleAttr) {
+    private val data = MutableStateFlow<PaymentMethodMessagingData?>(null)
+    private var job: Job? = null
+    private val stripeImageLoader = StripeImageLoader(context)
+
+    @Composable
+    override fun Content() {
+        data.collectAsState().value?.let { data ->
+            PaymentMethodMessaging(
+                data = data
+            )
+        }
+    }
+
+    fun load(
+        config: Configuration,
+        onSuccess: () -> Unit,
+        onFailure: (Throwable) -> Unit
+    ) {
+        val viewModel: PaymentMethodMessagingViewModel = ViewModelProvider(
+            context as ViewModelStoreOwner,
+            PaymentMethodMessagingViewModel.Factory { config }
+        )[PaymentMethodMessagingViewModel::class.java]
+        job = CoroutineScope(Dispatchers.IO).launch {
```

Comment:
This feels like something we should do in the view model.

If I were to reimagine this, we'd call a load method on the view model, and observe a stateflow in the view.

Doing it this way would closer follow android lifecycle best practices. 

Happy to talk in more depth, or pair on this if you're interested.